test/docker_container_test_case.py

```diff
@@ -90,13 +95,26 @@ def _create_container(cls, image: str, container_port: int, **kwargs) -> Netloc:
                 log.info('Launched sibling container %s from image %s, listening on %s:%i',
                          container.name, image, container_ip, container_port)
             else:
+                retries = 0
                 ports = network_settings['Ports']
+                while len(ports[f'{container_port}/tcp']) < 1:
+                    if (datetime.now() - time_it).seconds > 3:
+                        # Wait for the published ports of the container that's
+                        # supposedly running, otherwise giveup (let it fail).
+                        log.error('Unreachable TCP port %s for container %s',
+                                  container_port, container.name)
+                        break
```

Comment:
Why postpone the failure? Raising an exception here makes both the source code and the resulting stack trace easier to understand.